lib/js_routes.rb

```diff
@@ -117,19 +118,26 @@ def generate
       application.reload_routes!
     end
 
+    deprecated_globbing_behavior = ActionPack::VERSION::MAJOR == 4 && ActionPack::VERSION::MINOR == 0
+
     {
       "GEM_VERSION"         => JsRoutes::VERSION,
       "ROUTES"              => js_routes,
       "NODE_TYPES"          => json(NODE_TYPES),
       "RAILS_VERSION"       => ActionPack.version,
-      "DEPRECATED_GLOBBING_BEHAVIOR" => ActionPack::VERSION::MAJOR == 4 && ActionPack::VERSION::MINOR == 0,
-
-      "APP_CLASS"           => application.class.to_s,
-      "NAMESPACE"           => json(@configuration.namespace),
-      "DEFAULT_URL_OPTIONS" => json(@configuration.default_url_options),
-      "PREFIX"              => json(@configuration.prefix),
-      "SPECIAL_OPTIONS_KEY" => json(@configuration.special_options_key),
-      "SERIALIZER"          => @configuration.serializer || json(nil),
+      "DEPRECATED_GLOBBING_BEHAVIOR" => deprecated_globbing_behavior,
+
+      "APP_CLASS"                 => application.class.to_s,
+      "NAMESPACE"                 => json(@configuration.namespace),
+      "DEFAULT_URL_OPTIONS"       => json(@configuration.default_url_options),
+      "PREFIX"                    => json(@configuration.prefix),
+      "SPECIAL_OPTIONS_KEY"       => json(@configuration.special_options_key),
+      "SERIALIZER"                => @configuration.serializer || json(nil),
+      "URI_ENCODER_SEGMENT_REGEX" => (JsRegex.new(
```

Comment:
I think this new dependency can be development dependency but not runtime dependency (just like coffee script): we can compile regexps only once when the gem is "built".
           
I can even think about hardcode if the dynamic implementation would be too complex: the regexp was last updated in 2014 and I don't think it will ever be updated again. https://github.com/rails/rails/commit/5460591f0226a9d248b7b4f89186bd5553e7768f